### PR TITLE
[backend]Verify createdBy type on malware creation (#6291)

### LINKED ISSUE #6291: Creation of created-by relatiohship is not protected to target only identity
## Description
Creation of created-by relationship is not protected to target only identity.
At creation it will not fail but then the read will fail.

### Steps
1. Get a user id from the platform
2. Create a malware + read
```
mutation createMalware {
  malwareAdd(input: {
    name: "new_malware",
    createdBy: "dc5f1601-e4ff-4d17-9233-ec80ceef207d"
  }) {
    id
    createdBy {
      id
    }
  }
}
```

### Result
Create ok but return read fail.
`"message": "Runtime Object type \"User\" is not a possible type for \"Identity\".",`

### Expected
The creation must fail has USER is not a correct target for createdBy.

### PR DESCRIPTION
**Pull request by aHenryJard**
### Proposed changes

* Verify that the id given in createdBy relation is an Identity, if not throw exception and prevent from creating entity.
  * on creation
  * on update
  * on upsert
  * on relationAdd
* Test are written using malware use case, but the implementation is for all entity that have createdBy relationship
* Note that some changes in test (mainly '?' operator) are because of refactor of file from js to ts.

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/6291

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [x] I wrote test cases for the relevant uses case
- [x] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/malware-test.ts [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+      mutation MalwareAdd($input: MalwareAddInput!) {
+        malwareAdd(input: $input) {
+          name
+          createdBy {
+            id
+          }
+        }
+      }
+    `;
+    const MALWARE_WITH_BAD_CREATED_BY = {
```
Comment (JeremyCloarec) [inline opencti-platform/opencti-graphql/tests/02-integration/02-resolvers/malware-test.ts]: small typo when copy/pasting from previous test, we can name it MALWARE_WITH_VALID_CREATED_BY

File: opencti-platform/opencti-graphql/src/database/middleware.js [hunk 2]
```diff
@@ -144,7 +145,7 @@ import {
   isStixDomainObjectShareableContainer,
   isStixObjectAliased,
   resolveAliasesField,
-  STIX_ORGANIZATIONS_UNRESTRICTED,
+  STIX_ORGANIZATIONS_UNRESTRICTED
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-graphql/src/database/middleware.js]: why do we have this change ? the last ',' should be kept

File: opencti-platform/opencti-graphql/src/database/middleware.js [hunk 3]
```diff
@@ -565,7 +566,33 @@ const idLabel = (label) => {
...
+ * @param context
+ * @param user
+ * @param createdById
+ * @returns {Bluebird.Promise<void>}
+ */
+export const validateCreatedBy = async (context, user, createdById) => {
+  if (createdById) {
+    const createdByEntity = await internalLoadById(context, user, createdById);
+    if (createdByEntity && createdByEntity.parent_types) {
+      if (!createdByEntity.parent_types.some((parent) => parent === ENTITY_TYPE_IDENTITY)) {
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-graphql/src/database/middleware.js]: I wonder if you could you use `isStixDomainObjectIdentity(createdByEntity.entity_type)` ?

File: opencti-platform/opencti-graphql/src/database/middleware.js (hunk 2 above)
Comment (aHenryJard) [reply opencti-platform/opencti-graphql/src/database/middleware.js]: I think it's ESLint that did it, I can revert that

File: opencti-platform/opencti-graphql/src/database/middleware.js (hunk 3 above)
Comment (aHenryJard) [reply opencti-platform/opencti-graphql/src/database/middleware.js]: you're right, it's working like a charm, thanks !

**Review by JeremyCloarec — APPROVED** [review]

File: opencti-platform/opencti-graphql/src/database/middleware.js (hunk 2 above)
Comment (JeremyCloarec) [reply opencti-platform/opencti-graphql/src/database/middleware.js]: I'm not sure I understand what's the convention for trailing commas in imports? There is already a mix of imports having a trailing comma on their last import line, and others not having it

**Pull request merged**